### PR TITLE
move accountProvider to redux, drop proxy and Api provider

### PR DESCRIPTION
**Pull request by StanislavSava**

### CODE REVIEW

File: src/apiCalls/accounts/getAccountFromApi.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import { store } from 'reduxStore/store';
+import { addressEndpoint } from 'apiCalls/endpoints';
+import { AccountType } from 'types';
+
+export async function getAccountFromApi(address?: string) {
+  if (!address) {
+    return null;
+  }
+  const apiAddress = apiAddressSelector(store.getState());
+  const cleanApiAddress = apiAddress.endsWith('/')
```
Comment (grigorebudac) [inline src/apiCalls/accounts/getAccountFromApi.ts:12]: it seems like this part is being copy-pasted in multiple files. I would create a new selector here, something like cleanApiAddressSelector

File: src/hooks/transactions/useSignTransactions.tsx [hunk 2]
```diff
@@ -149,7 +159,7 @@ export const useSignTransactions = () => {
       }
     } catch (error) {
       const errorMessage =
-        (error as unknown as Error)?.message ||
+        ((error as unknown) as Error)?.message ||
```
Comment (grigorebudac) [inline src/hooks/transactions/useSignTransactions.tsx:162]: have you tried casting it directly as Error?

File: src/hooks/transactions/useSignTransactions.tsx [hunk 3]
```diff
@@ -217,7 +216,7 @@ export const useSignTransactions = () => {
         signTransactionsWithProvider();
       }
     } catch (err) {
-      const defaultErrorMessage = (error as unknown as Error)?.message;
+      const defaultErrorMessage = ((error as unknown) as Error)?.message;
```
Comment (grigorebudac) [inline src/hooks/transactions/useSignTransactions.tsx:219]: same here

File: src/reduxStore/store.ts [hunk 4]
```diff
@@ -51,9 +51,14 @@ export const store = configureStore({
           PURGE,
           REGISTER,
           'accountInfoSlice/setAccount',
-          'accountInfoSlice/setAccountNonce'
+          'accountInfoSlice/setAccountNonce',
```
Comment (grigorebudac) [inline src/reduxStore/store.ts:54]: this approach is ok, alternatively, you can get the name dynamically from the slice

Comment (arhtudormorar) [inline src/reduxStore/store.ts:54]: would be nice to use these strings from a config

File: src/types/network.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  walletConnectDeepLink?: string;
+  walletConnectBridgeAddresses?: string[];
+  walletAddress?: string;
+  apiAddress?: string;
+  explorerAddress?: string;
+  skipFetchFromServer?: boolean;
+  apiTimeout?: string;
+}
+
+export interface BlockchainNetworkConfigType {
```
Comment (arhtudormorar) [inline src/types/network.ts:41]: rename to ApiNetworkConfigType

File: src/wrappers/DappProvider/index.tsx [hunk 6]
```diff
@@ -21,8 +17,6 @@ export interface DappProviderPropsType {
   customNetworkConfig?: CustomNetworkType;
   externalProvider?: IDappProvider;
   environment: 'testnet' | 'mainnet' | 'devnet' | EnvironmentsEnum;
```
Comment (arhtudormorar) [inline src/wrappers/DappProvider/index.tsx:19]: can use `keyof typeof EnvironmentsEnum`

**Review by cipriandraghici93 — APPROVED** [review]

**Pull request merged**